### ISSUE TITLE
ArrayIndexOutOfBoundsException: length=1; index=1

### ISSUE THREAD

**Issue by TheSiebi** [tensorflow/tensorflow#28815]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10, Android Studio 3.4.1
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
OnePlus 6

**Describe the problem**

I am using the tensorflow example object_detection app, it's working fine. But when I try plugging in my own model "tiny-yolo-obj.lite" (created with Tiny YOLO) and "labelmap2.txt" (first line being ???, other 36 lines my objects) it throws the following error:

```
Process: org.tensorflow.lite.examples.detection, PID: 31738
    java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
```

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**My model meta:**

`{"net": {"type": "[net]", "batch": 64, "subdivisions": 8, "width": 416, "height": 416, "channels": 3, "momentum": 0.9, "decay": 0.0005, "angle": 0, "saturation": 1.5, "exposure": 1.5, "hue": 0.1, "learning_rate": 0.001, "max_batches": 40200, "policy": "steps", "steps": "-1,100,20000,30000", "scales": ".1,10,.1,.1"}, "type": "[region]", "anchors": [1.08, 1.19, 3.42, 4.41, 6.63, 11.38, 9.42, 5.11, 16.62, 10.52], "bias_match": 1, "classes": 36, "coords": 4, "num": 5, "softmax": 1, "jitter": 0.2, "rescore": 1, "object_scale": 5, "noobject_scale": 1, "class_scale": 1, "coord_scale": 1, "absolute": 1, "thresh": 0.6, "random": 1, "model": "cfg/tiny-yolo-obj.cfg", "inp_size": [416, 416, 3], "out_size": [13, 13, 205], "name": "tiny-yolo-obj", "labels": ["Rosen 6", "Rosen 7", "Rosen 8", "Rosen 9", "Rosen 10", "Rosen Under", "Rosen Ober", "Rosen K\u00f6nig", "Rosen Ass", "Eichel 6", "Eichel 7", "Eichel 8", "Eichel 9", "Eichel 10", "Eichel Under", "Eichel Ober", "Eichel K\u00f6nig", "Eichel Ass", "Schellen 6", "Schellen 7", "Schellen 8", "Schellen 9", "Schellen 10", "Schellen Under", "Schellen Ober", "Schellen K\u00f6nig", "Schellen Ass", "Schilten 6", "Schilten 7", "Schilten 8", "Schilten 9", "Schilten 10", "Schilten Under", "Schilten Ober", "Schilten K\u00f6nig", "Schilten Ass"], "colors": [[254.0, 254.0, 254], [246.0625, 222.25, 127], [238.125, 190.5, 0], [230.1875, 158.75, -127], [222.25, 127.0, 254], [214.3125, 95.25, 127], [206.375, 63.5, 0], [198.4375, 31.75, -127], [190.5, 0.0, 254], [182.5625, -31.75, 127], [174.625, -63.5, 0], [166.6875, -95.25, -127], [158.75, -127.0, 254], [150.8125, -158.75, 127], [142.875, -190.5, 0], [134.9375, -222.25, -127], [127.0, 254.0, 254], [119.0625, 222.25, 127], [111.125, 190.5, 0], [103.1875, 158.75, -127], [95.25, 127.0, 254], [87.3125, 95.25, 127], [79.375, 63.5, 0], [71.4375, 31.75, -127], [63.5, 0.0, 254], [55.5625, -31.75, 127], [47.625, -63.5, 0], [39.6875, -95.25, -127], [31.75, -127.0, 254], [23.8125, -158.75, 127], [15.875, -190.5, 0], [7.9375, -222.25, -127], [0.0, 254.0, 254], [-7.9375, 222.25, 127], [-15.875, 190.5, 0], [-23.8125, 158.75, -127]]}`

**My code in DetectorActivity**

```
  // Configuration values for the prepackaged SSD model.
  private static final int TF_OD_API_INPUT_SIZE = 300;
  private static final boolean TF_OD_API_IS_QUANTIZED = false; // maybe also true, dunno yet
  private static final String TF_OD_API_MODEL_FILE = "tiny-yolo-obj.lite";
  private static final String TF_OD_API_LABELS_FILE = "labelmap2.txt";
  private static final DetectorMode MODE = DetectorMode.TF_OD_API;
  // Minimum detection confidence to track a detection.
  private static final float MINIMUM_CONFIDENCE_TF_OD_API = 0.5f;
  private static final boolean MAINTAIN_ASPECT = false;
  private static final Size DESIRED_PREVIEW_SIZE = new Size(640, 480);
  private static final boolean SAVE_PREVIEW_BITMAP = false;
  private static final float TEXT_SIZE_DIP = 10;
```

Any help would be greatly appreciated! Thanks in advance.

**Comment by TheSiebi** [comment]
Issue resolved, I had to use DetectorMode.YOLO

**Comment by achandraa** [comment]
Good to know the issue is resolved and thank you for mentioning the resolution

**Comment by ymodak** [comment]
Closing this issue since its resolved. Thanks!

**Comment by Ehtasha** [comment]
@TheSiebi   @ymodak  
I'm integrating MakeMl nails model in android app and facing same issue.

Error Message: java.lang.ArrayIndexOutOfBoundsException: length=66049; index=66049

Model tflite File: https://github.com/makeml-app/MakeML-Nails/tree/master/Segmentation%20Nails/Resources

Android Code: https://github.com/RobinHan24/android-segmentation-app

Any suggestion. How I resolved this issue ?
